### PR TITLE
Move to generated keys in RSA tests

### PR DESCRIPTION
**Pull request by Groenbech96**
#### Summary 
Add methods to generate XML keys. Use new method to generate keys needed in tests.

#### Work Item(s) 
Fixes [AB#500563](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/500563)

### CODE REVIEW

File: src/System Application/App/Cryptography Management/src/RSACryptoServiceProviderImpl.Codeunit.al [hunk 1]
```diff
@@ -32,6 +32,17 @@ codeunit 1446 "RSACryptoServiceProvider Impl." implements "Signature Algorithm v
...
 
+    [NonDebuggable]
+    procedure CreateRSAKeyPair(var PublicKeyInXML: Text; var PrivateKeyInXML: SecretText)
+    var
+        DotnetRSA: DotNet RSA;
+    begin
+        RSACryptoServiceProvider();
+        DotnetRSA := DotNetRSACryptoServiceProvider.Create();
+        PublicKeyInXML := DotnetRSA.ToXmlString(false);
+        PrivateKeyInXML := DotnetRSA.ToXmlString(true);
```
Comment (AndreasMoth) [inline src/System Application/App/Cryptography Management/src/RSACryptoServiceProviderImpl.Codeunit.al]: From what I can read, this is not just the private key but both the public and private key: https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.rsa.toxmlstring?view=net-8.0

File: src/System Application/App/Cryptography Management/src/RSACryptoServiceProvider.Codeunit.al [hunk 2]
```diff
@@ -155,4 +155,17 @@ codeunit 1445 RSACryptoServiceProvider
...
+
+    /// <summary>  
+    /// The CreateRSAKeyPair procedure is a function that generates a public and private RSA key pair.  
+    /// </summary>  
+    /// <param name="PublicKeyInXML">This is an output parameter that returns the public key in XML format.</param>  
+    /// <param name="PrivateKeyInXML">This is an output parameter that returns the private key in XML format. This is a sensitive information hence marked as SecretText.</param>  
+    /// <returns>  
+    /// This function does not return a value. The output is via the two parameters PublicKeyInXML and PrivateKeyInXML.  
+    /// </returns>  
+    procedure CreateRSAKeyPair(var PublicKeyInXML: Text; var PrivateKeyInXML: SecretText)
```
Comment (AndreasMoth) [inline src/System Application/App/Cryptography Management/src/RSACryptoServiceProvider.Codeunit.al]: Pascal casing? PublicKeyInXml

File: src/System Application/App/Cryptography Management/src/RSACryptoServiceProviderImpl.Codeunit.al (hunk 1 above)
Comment (Groenbech96) [reply src/System Application/App/Cryptography Management/src/RSACryptoServiceProviderImpl.Codeunit.al]: @AndreasMoth depends on what you give as the parameter. See includePrivateParameters in the link you provided.

Comment (AndreasMoth) [reply src/System Application/App/Cryptography Management/src/RSACryptoServiceProviderImpl.Codeunit.al]: Yeah, that's the one I'm referencing:
image
Not a big deal though, just that it contains both :)

**Review by nikolakukrika — DISMISSED** [review]

File: src/System Application/App/Cryptography Management/src/RSACryptoServiceProviderImpl.Codeunit.al (hunk 1 above)
Comment (Groenbech96) [reply src/System Application/App/Cryptography Management/src/RSACryptoServiceProviderImpl.Codeunit.al]: The PrivateKeyXmlStringSecret used in the tests is containing exactly that, both pub and private. That is how their implementation works, also when you call FromXMLString().

Comment (darjoo) [reply src/System Application/App/Cryptography Management/src/RSACryptoServiceProviderImpl.Codeunit.al]: This is a non-issue. Resolving.

**Review by darjoo — APPROVED** [review]

**Review by WaelAbuSeada — APPROVED** [review]

**Review by JesperSchulz — APPROVED** [review]

**Pull request merged**